### PR TITLE
BLD: Build arm64 wheels on CircleCI

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Draft pending #48945 merge.
Uploading also needs to be figuring out.
Most likely will use the Trigger CircleCI action, passing in the env vars.

### CODE REVIEW

**Comment by WillAyd** [comment]
Looks like the dependency is merged - is this ready to come out of draft status?

**Comment by datapythonista** [comment]
@lithomas1 should we run the CI for this, and see if it works as expected?

**Comment by lithomas1** [comment]
This PR works(sans uploading), but we need someone to add the Anaconda upload tokens to CircleCI.
(It's probably wise to hold off on that for now, while they sort out their security breach).

@TomAugspurger Can you add me to the scipy-wheels-nightly and multibuild-wheels-staging Anaconda orgs? I am also lithomas1 on anaconda.

**Comment by TomAugspurger** [comment]
@lithomas1 I added you to both those.

**Comment by lithomas1** [comment]
OK, this is ready now, I will still have to add the token to CircleCI after this is merged, though.

File: .circleci/config.yml
Comment (mroeschke) [inline .circleci/config.yml:32]: Not a big deal, but is there a way to limit the fetch depth?

File: .circleci/config.yml
Comment (mroeschke) [inline .circleci/config.yml:26]: Does it make sense to bump to `ubuntu-2004:2022.04.1`?

File: .circleci/config.yml
Comment (mroeschke) [inline .circleci/config.yml:42]: Is it worth putting a retry on the `curl`?

File: .circleci/config.yml
Comment (lithomas1) [reply .circleci/config.yml:26]: I'll do it in a follow up. I'm currently matching the other arm64 job.

File: .circleci/config.yml
Comment (lithomas1) [reply .circleci/config.yml:32]: The checkout step can't be configured. We could use raw git commands, but I don't think its worth the time savings (we still need to pull tags).

File: .circleci/config.yml
Comment (lithomas1) [reply .circleci/config.yml:44]: I added specifying "wheel build" in the commit message as an alternative.

There's not much we can do if we get rate limited.

**Review by mroeschke — APPROVED** [review]
LGTM

**Comment by mroeschke** [comment]
I guess this is the first time testing Windows + ARM and looks like `test_ea_dtypes_and_scalar_operation` is failing.

Seems like this line should account for `is_platform_arm()`

```
[1, 1], dtype="int64" if not is_platform_windows() else "int32"
```

**Comment by lithomas1** [comment]
> I guess this is the first time testing Windows + ARM and looks like `test_ea_dtypes_and_scalar_operation` is failing.
> 
> Seems like this line should account for `is_platform_arm()`
> 
> ```
> [1, 1], dtype="int64" if not is_platform_windows() else "int32"
> ```

That's a Windows failure, only, and its failing on main too unfortunately, since recently. 

(I should probably document this more, but you find status of the nightly builds by going to Actions -> Wheel builder workflow -> Filter event by schedule).

From https://github.com/pandas-dev/pandas/actions/workflows/wheels.yml?query=event%3Aschedule, it's been failing since  yesterday.

I think one thing that would be nice would be to have the Wheel builders automatically open issues for failures (will add to my backlog).

**Comment by phofl** [comment]
This is currently running on my prs. Is this intended?

**Comment by lithomas1** [comment]
I messed up the grep expression (needed to escape brackets, PR incoming).

**Pull request merged**
